community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/BatchingMultipleIndexPopulator.java

```diff
@@ -131,14 +129,11 @@ protected IndexPopulation createPopulation( IndexPopulator populator,
     @Override
     protected void populateFromQueue( long currentlyIndexedNodeId )
     {
-        if ( queue.size() >= QUEUE_THRESHOLD )
-        {
-            log.debug( "Populating from queue." + EOL + this );
-            flushAll();
-            awaitCompletion();
-            super.populateFromQueue( currentlyIndexedNodeId );
-            log.debug( "Drained queue and all batched updates." + EOL + this );
-        }
+        log.debug( "Populating from queue." + EOL + this );
+        flushAll();
```

Comment:
Doing this flushing every time there's an update will produce more and smaller tasks to the executor, won't it? Could it be less efficient to do this? Have you measured?